pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -0,0 +1,201 @@
+package nstemplatetiers
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+	templatev1 "github.com/openshift/api/template/v1"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("templates")
+
+// CreateOrUpdateResources generates the NSTemplateTier resources from the namespace templates,
+// then uses the manager's client to create or update the resources on the cluster.
+func CreateOrUpdateResources(mgr manager.Manager, namespace string, asset func(name string) ([]byte, error)) error {
+	g, err := NewNSTemplateTierGenerator(mgr.GetScheme(), asset)
+	if err != nil {
+		return errors.Wrap(err, "unable to create or update NSTemplateTiers")
+	}
+	tiers, err := g.NewNSTemplateTiers(namespace)
+	if err != nil {
+		return errors.Wrap(err, "unable to create or update NSTemplateTiers")
+	}
+	for _, tier := range tiers {
+		mgr.GetClient().Create(context.TODO(), tier)
+	}
+	return nil
+}
+
+// NSTemplateTierGenerator the NSTemplateTier manifest generator
+type NSTemplateTierGenerator struct {
+	asset     func(name string) ([]byte, error) // the func which gives access to the
+	revisions map[string]map[string]string
+	decoder   runtime.Decoder
+}
+
+// NewNSTemplateTierGenerator returns a new NSTemplateTierGenerator
+func NewNSTemplateTierGenerator(s *runtime.Scheme, asset func(name string) ([]byte, error)) (*NSTemplateTierGenerator, error) {
+	metadata, err := asset("metadata.yaml")
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to initialize the NSTemplateTierGenerator")
```

Comment:
```suggestion
		return nil, errors.Wrapf(err, "unable to get the metadata.yaml asset while initializing NSTemplateTierGenerator")
```

Reply:
ok, this change may make sense, although, I believe that the `Asset` func would return an error which mentions the `metadata.yaml` file